### PR TITLE
bugfix for high frequency time series in scaling.py, regarding issue …

### LINKED ISSUE #1257: Error in scaling.py when sampling rate faster than 1s.
**Describe the bug**
`scaling._compute_wavelet(clearsky_index)` produces `OverflowError` when passing it a `pandas.Series` type with a time resolution less than one second. Bug occurs due to the way dt is calculated for pandas types. Using pd.timedelta.seconds coerces to an integer seconds, resulting in value of zero. Produces infinity later when we take the log of dt.

Easiest fix could be:
`202 dt = (clearsky_index.index[1] - clearsky_index.index[0]).seconds + (clearsky_index.index[1] - clearsky_index.index[0]).microseconds/1e6`

Will write test for this case and submit pull request.

Current Workaround if anybody is actually running into this: 
convert Series to np.array and manually specify dt in call. Example when `kt` is a `pandas.Series`:
`pvlib.scaling.wvm(np.array(kt), positions, cloud_speed, dt=0.01)`

### PR DESCRIPTION
**Pull request by jranalli**
…1257

 - [ X ] Closes #1257
 - [ X ] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ X ] Tests added
 - [ X ] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [ X ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ X ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [ X ] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

When passing pvlib.scaling._compute_wavelet a pandas timeseries with a time resolution faster than 1s, the use of .seconds to calculate the time series delta_t produced an overflow error. It occurs because .seconds only computes integer values of seconds. I added a computation including microseconds to account for fractional seconds in this case.

### CODE REVIEW

**Comment by jranalli** [comment]
I'm going to push an additional commit proposing a slight refactor to scaling.wvm. I have a use case for having a separate computation of the variability reduction factors used by the WVM. In a recent conference paper I was looking at how to apply the WVM in reverse to increase the spatial resolution. While I'm not proposing we include that reverse case (until such time as someone is able to generate more comprehensive published results for it), having a computation of the VR available would make it possible to do that calculation on your own with minimal re-writing of WVM code to perform those calculations. It has no impact on the existing WVM calculation, but I did write a test for the proposed new function.

If putting this change here is inappropriate, or this is an unwanted change, we can just unwind this commit and stick with the bugfix above.

**Comment by cwhanse** [comment]
The refactor makes the main function more readable, IMO. I'd keep it and if it supports future capability, all the better.

**Comment by jranalli** [comment]
Sounds good, I will fix the stickler errors and squash the commits then.

File: docs/sphinx/source/whatsnew/v0.9.0.rst [hunk 1]
```diff
@@ -187,6 +187,10 @@ Bug fixes
 * Corrected an error affecting :py:func:`~pvlib.clearsky.detect_clearsky`
   when data time step is not one minute. Error was introduced in v0.8.1.
   (:issue:`1241`, :pull:`1242`)
+* Corrected error affecting :py:func:`~pvlib.scaling._compute_wavelet` when
+  passing a pandas time series with a sampling rate faster than 1 second.
+  (:issue:`1257`, :pull:`1258`)
+  (:issue:`1257`, :pull:``)
```
Comment (cwhanse) [inline docs/sphinx/source/whatsnew/v0.9.0.rst:193]: Delete extra line

**Review by cwhanse — APPROVED** [review]
@jranalli can you remove the extra line in the whatsnew file?

**Comment by jranalli** [comment]
Sorry about that, I think I got it.

**Comment by wholmgren** [comment]
@cwhanse can you merge if this is ready?

**Comment by cwhanse** [comment]
thanks @jranalli

**Comment by jranalli** [comment]
No problem, thanks for including it!

**Pull request merged**